### PR TITLE
LOG4J-1997 Cleanup of deprecated methods, warnings and use of try-with-resource to

### PR DESCRIPTION
**Pull request by kpm1985**
remove resource leaks.

### CODE REVIEW

File: log4j-core/src/test/java/org/apache/logging/log4j/core/appender/HangingAppender.java [hunk 1]
```diff
@@ -90,4 +90,8 @@ public boolean stop(final long timeout, final TimeUnit timeUnit) {
         setStopped();
         return true;
     }
+
+	public static long getSerialversionuid() {
```
Comment (mikaelstaldal) [inline log4j-core/src/test/java/org/apache/logging/log4j/core/appender/HangingAppender.java]: What's the point of this?

Comment (garydgregory) [inline log4j-core/src/test/java/org/apache/logging/log4j/core/appender/HangingAppender.java]: None that I can think of...

**Comment by kpm1985** [comment]
Remove eclipse warning for the unused field unless you think the getter is
a big deal

**Comment by mikaelstaldal** [comment]
Such warning is bogus, try to disable it rather than adding an unnecessary method to the code base.

**Comment by garydgregory** [comment]
Agreed.

**Comment by kpm1985** [comment]
Done

**Comment by kpm1985** [comment]
Please let me know if you have any other feedback, I have removed the useless getter method.

**Comment by kpm1985** [comment]
I've also made some more changes to reflect consistent non wildcard imports, spacing of imports, etc that makes the core test package look a little cleaner.

**Comment by kpm1985** [comment]
I will create a new patch and pull request that addresses the minimal set of changes to close LOG4J2-1997. Although the changes are fairly trivial, I realize that 27 files might be too large. Thank you for feedback on my previous PR.

**Pull request closed without merging**